src/dotnet/Search.Service/Module/SearchServiceModule.cs

```diff
@@ -60,7 +60,7 @@ protected override void InjectServices(IServiceCollection services)
             new SingleNodeConnectionPool(new Uri(openSearchClusterUri)),
             sourceSerializer: (builtin, settings) => new OpenSearchJsonSerializer(builtin, settings));
         services.AddSingleton<IOpenSearchClient>(_ => new OpenSearchClient(connectionSettings));
-        services.AddSingleton<OpenSearchConfigurator>()
+        services.AddSingleton(s => ActivatorUtilities.CreateInstance<OpenSearchConfigurator>(s, HostInfo.IsDevelopmentInstance))
```

Comment:
Can we use HostInfo inside OpenSearchConfigurator?

Reply:
done